flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -0,0 +1,1057 @@
+import org.apache.flink.table.calcite._
+import org.apache.flink.table.catalog.{CatalogBaseTable, CatalogManager, CatalogManagerCalciteSchema, ConnectorCatalogTable, FunctionCatalog, GenericInMemoryCatalog, ObjectPath, QueryOperationCatalogView}
+import org.apache.flink.table.dataformat.BaseRow
+import org.apache.flink.table.expressions.TableReferenceExpression
+import org.apache.flink.table.expressions.resolver.lookups.TableReferenceLookup
+import org.apache.flink.table.functions.utils.UserDefinedFunctionUtils.{checkForInstantiation, checkNotSingleton, extractResultTypeFromTableFunction, getAccumulatorTypeOfAggregateFunction, getResultTypeOfAggregateFunction}
+import org.apache.flink.table.functions.{AggregateFunction, ScalarFunction, TableFunction}
+import org.apache.flink.table.operations.utils.OperationTreeBuilder
+import org.apache.flink.table.operations.{CatalogQueryOperation, PlannerQueryOperation, QueryOperation}
+import org.apache.flink.table.plan.nodes.calcite.{LogicalSink, Sink}
+import org.apache.flink.table.plan.nodes.exec.ExecNode
+import org.apache.flink.table.plan.nodes.physical.FlinkPhysicalRel
+import org.apache.flink.table.plan.optimize.Optimizer
+import org.apache.flink.table.plan.reuse.SubplanReuser
+import org.apache.flink.table.plan.util.SameRelObjectShuttle
+import org.apache.flink.table.planner.PlannerContext
+import org.apache.flink.table.sinks.TableSink
+import org.apache.flink.table.sources.TableSource
+import org.apache.flink.table.types.LogicalTypeDataTypeConverter.fromDataTypeToLogicalType
+import org.apache.flink.table.types.logical.{LogicalType, RowType, TypeInformationAnyType}
+import org.apache.flink.table.types.utils.TypeConversions.fromLegacyInfoToDataType
+import org.apache.flink.table.types.{ClassLogicalTypeConverter, DataType, TypeInfoDataTypeConverter}
+import org.apache.flink.table.typeutils.TimeIndicatorTypeInfo
+import org.apache.flink.table.util.JavaScalaConversionUtil
+import org.apache.flink.types.Row
+
+import org.apache.calcite.jdbc.CalciteSchemaBuilder.asRootSchema
+import org.apache.calcite.plan.{RelTrait, RelTraitDef}
+import org.apache.calcite.rel.RelNode
+import org.apache.calcite.sql._
+import org.apache.calcite.tools._
+
+import _root_.java.lang.reflect.Modifier
+import _root_.java.util
+import _root_.java.util.concurrent.atomic.AtomicInteger
+
+import _root_.scala.annotation.varargs
+import _root_.scala.collection.JavaConversions._
+import _root_.scala.collection.JavaConverters._
+import _root_.scala.collection.mutable
+
+/**
+  * The abstract base class for batch and stream TableEnvironments.
+  *
+  * @param streamEnv The [[JavaStreamExecEnv]] which is wrapped in this
+  *                [[StreamTableEnvironment]].
+  * @param config The configuration of the TableEnvironment
+  * @param catalogManager a catalog manager that encapsulates all available catalogs.
+  */
+abstract class TableEnvironment(
```

Comment:
What this file for?
I think it has been removed.

Reply:
Yes, it's due to a bad rebase-to-master. It's been removed